### PR TITLE
Save and return tests

### PR DESCRIPTION
**Pull request by tommotaylor**
Automated tests for new save and return functionality.

Does not test 2FA, that will be a different PR

### CODE REVIEW

File: tests/save-and-return.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import withPage from '../utils/withPage'
+import config from '../config'
+
+test(
+  'Save & return button is displayed to the user',
+  withPage,
+  async (t, page) => {
+    await page.goto(config.formURL)
+    t.is(
+      await page.getText('a[href=\'/return\']'),
```
Comment (solidgoldpig) [inline tests/save-and-return.js]: Can we make the sign-in link be a variable created in config?

File: tests/save-and-return.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import test from 'ava'
+import withPage from '../utils/withPage'
+import config from '../config'
+
+test(
+  'Save & return button is displayed to the user',
```
Comment (solidgoldpig) [inline tests/save-and-return.js]: How are we testing that save and return button is not being displayed to user? Perrsonally I think we need more than one form...

File: tests/save-and-return.js (hunk 1 above)
Comment (tommotaylor) [reply tests/save-and-return.js]: Still a WIP

File: tests/save-and-return.js (hunk 2 above)
Comment (tommotaylor) [reply tests/save-and-return.js]: Good point. Either another form or a way to change settings on the fly. We now have two settings... 1) save & return, 2) user attachments, that need to be turned on and off for testing each one.

File: package.json [hunk 3]
```diff
@@ -5,7 +5,8 @@
   "description": "End to end tests for a Form Builder form",
   "scripts": {
     "lint": "eslint .",
-    "test": "npm run lint && ava --reset-cache && ava -v",
+    "test": "ava --reset-cache && ava -v",
```
Comment (solidgoldpig) [inline package.json]: Why remove linting from test?

File: utils/email-service.js [hunk 4]
```diff
@@ -13,8 +13,8 @@ function waitForEmail (recipientEmail) {
   })
 }
 
-function generateRandomEmailAddress () {
-  return client.servers.generateEmailAddress(serverID)
+function generateEmailAddress (subdomain = 'xyz') {
+  return `${subdomain}.${process.env.EMAIL_SERVICE_SERVER_ID}@mailosaur.io`
```
Comment (solidgoldpig) [inline utils/email-service.js]: What's the reason for this change?

Comment (tommotaylor) [reply utils/email-service.js]: A few reasons:
1) It was named "generateRandom.." - it wasn't actually random
2) Because it used the same email address each time, when more than one test was sending emails, sometimes the wrong email was being retrieved.
3) By being able to change the email address per spec, we can ensure that we are only retrieving the correct emails.

You can see how we use this in the various specs, eg: `generateEmailAddress('save-form')`

Comment (solidgoldpig) [reply utils/email-service.js]: 👍

File: package.json (hunk 3 above)
Comment (solidgoldpig) [reply package.json]: 👍 I'd have restructured the circle-ci stuff as I prefer the main `test` to run everything, but that's just me…
